app/main.py

```diff
@@ -1,14 +1,67 @@
-class Validator:
-    pass
+from abc import ABC, abstractmethod
+from typing import Any
 
 
-class Number:
-    pass
+class Validator(ABC):
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.protected_name = "_" + name
 
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        return getattr(instance, self.protected_name)
 
-class OneOf:
-    pass
+    def __set__(self, instance: Any, value: int | str) -> None:
+        self.validate(value)
+        setattr(instance, self.protected_name, value)
+
+    @abstractmethod
+    def validate(self, value: int | str) -> None:
+        pass
+
+
+class Number(Validator):
+    def __init__(self, min_value: int, max_value: int) -> None:
+        self.min_value = min_value
+        self.max_value = max_value
+
+    def validate(self, value: int) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Quantity should be integer.")
+        if value > self.max_value or value < self.min_value:
```

Comment:
You could write this check without using `or`, e.g. `if not self.min_value <= value <= self.max_value`.